### PR TITLE
Added logic to check for changes in org variable in msg object

### PR DESCRIPTION
**Pull request by Steve-Walsh**
Ticket: https://servisbot.atlassian.net/browse/MVP-6967

Just adding a non blocking log for any change in organization for now

image

### CODE REVIEW

File: nodes/StopTheBleed.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+const clone = require('clone');
+
+const variablesToCheck = [
+  'logger.metadata.organization',
+  'payload.system.organization',
+  'event.event.organization',
+  'event.event.token.contents.organization'
+];
+
+module.exports = class StopTheBleed {
```
Comment (johnrellis) [inline nodes/StopTheBleed.js]: rename

File: package.json [hunk 2]
```diff
@@ -37,7 +38,7 @@
     "bcryptjs": "2.4.3",
     "body-parser": "1.18.3",
     "cheerio": "0.22.0",
-    "clone": "2.1.1",
+    "clone": "^2.1.1",
```
Comment (johnrellis) [inline package.json]: why are we changing this?

**Review by johnrellis — CHANGES_REQUESTED** [review]
small comments

File: nodes/core/core/80-function.js [hunk 3]
```diff
@@ -207,10 +209,13 @@ module.exports = function(RED) {
         try {
             this.on("input", function(msg) {
                 try {
+                    const stopTheBleed = new StopTheBleed(msg)
                     var start = process.hrtime();
                     sandbox.msg = msg;
                     const vm2Instance = new vm2.VM({ sandbox, timeout: 5000 });
                     const result = vm2Instance.run(functionText);
+                    console.log('before the bleed check')
```
Comment (johnrellis) [inline nodes/core/core/80-function.js]: do you want this console.log?

Comment (Steve-Walsh) [reply nodes/core/core/80-function.js]: Nope, gone

File: package.json (hunk 2 above)
Comment (Steve-Walsh) [reply package.json]: Ah did an npm i of clone, didnt spot it in the list, will change back any remove the `^`

File: nodes/PayloadValidator.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  'logger.metadata.organization',
+  'payload.system.organization',
+  'event.event.organization',
+  'event.event.token.contents.organization'
+];
+
+module.exports = class PayloadValidator {
+  constructor(_before) {
+    const before = clone(_before);
+    const {
```
Comment (johnrellis) [inline nodes/PayloadValidator.js]: if there any way this deconstruction could fail?

File: nodes/PayloadValidator.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    this.bot = bot;
+    this.conversationId = conversationId;
+    this.organization = organization;
+    this.region = region;
+  }
+
+  verify(after) {
+    try {
+      variablesToCheck.forEach((location) => {
+        const getValue = (object) => location.split('.').reduce((p, c) => (p && p[c]) || null, object);
```
Comment (johnrellis) [inline nodes/PayloadValidator.js]: can this be in the global scope?

Comment (Steve-Walsh) [reply nodes/PayloadValidator.js]: Sure, just done it here so didn't have to pass around the location and object.

File: nodes/PayloadValidator.js (hunk 4 above)
Comment (Steve-Walsh) [reply nodes/PayloadValidator.js]: Only if they've deleted the items in the flow. I can add some safety around it

**Review by johnrellis — APPROVED** [review]

**Pull request merged**